ui/components/app/connected-sites-list/connected-snaps.js

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import { WALLET_SNAP_PERMISSION_KEY } from '@metamask/snaps-utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { SnapCaveatType } from '@metamask/rpc-methods';
+import { Box, IconName, IconSize, Text } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { MenuItem } from '../../ui/menu';
+import { SNAPS_VIEW_ROUTE } from '../../../helpers/constants/routes';
+import SnapAvatar from '../snaps/snap-avatar';
+import {
+  AlignItems,
+  BlockSize,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import ConnectedAccountsListOptions from '../connected-accounts-list/connected-accounts-list-options';
+import {
+  getOriginOfCurrentTab,
+  getPermissionSubjects,
+} from '../../../selectors';
+import { removePermissionsFor, updateCaveat } from '../../../store/actions';
+
+export default function ConnectedSnaps({ connectedSubjects }) {
+  const [showOptions, setShowOptions] = useState();
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const subjects = useSelector((state) => getPermissionSubjects(state));
```

Comment:
Can this just be: `useSelector(getPermissionSubjects)`?

Reply:
Yes https://github.com/MetaMask/metamask-extension/pull/20811/commits/e1739744968df5c9869c32187b166c2848c8dbf3